graphql/course/mutations.ts

```diff
@@ -106,6 +109,30 @@ export class MutateCourseResolver {
         }
         const result = await prisma.course.update({ data, where: { id: courseId } });
         logger.info(`Course (${result.id}) updated by Student (${context.user.studentId})`);
+
+        const subcourse = await prisma.subcourse.findFirst({
+            where: { courseId: courseId },
+        });
+        const usersSubcourseAchievements = await prisma.user_achievement.findMany({
+            where: { context: { path: ['relation'], equals: `subcourse/${subcourse.id}` }, userId: user.userID },
+            include: { template: true },
+        });
+        const subcourseAchievements = await Promise.all(
+            usersSubcourseAchievements.map(async (usersSubcourseAchievement) => {
+                return await prisma.user_achievement.findMany({
+                    where: { context: { path: ['relation'], equals: `subcourse/${subcourse.id}` }, templateId: usersSubcourseAchievement.template.id },
+                });
+            })
+        );
+        subcourseAchievements.flat().forEach(async (achievement) => {
```

Comment:
This should be awaited

Reply:
This was fixed by using a for ... of loop awaiting the prisma operations.